[DM-3888] [DM-3915] [DM-3989] - lwm2m change logs (#2440)
SoftwareAG/c8y-docs: content/change-logs/device-management/lwm2m-agent-10-20-385-0-improvement-empty-payload-no-ct.md

---
date:

> title: LWM2M agent can now handle an empty payload without content-type data

Reviewer: ```suggestion
title: Allowing LWM2M device to send empty payload without content-type data
```

change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
product_area: Device Management & Connectivity
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
ticket: DM-3989
version: 10.20.385.0
---

> The LWM2M specification permits devices to send an empty payload when there's no data to report. 

Reviewer 2: ```suggestion
The LWM2M specification allows devices to send an empty payload when there is no data to report. 
```

> However, it doesn’t specify whether a content-type should still be included in such cases. 

Reviewer 3: We are stating here that it is not specified which opens the question: how do we handle it now? If not specified, what is then the intended way to do it?

Author: We handle both cases

Author: Sorry - the intended way is that content type does not matter when there's no payload. It's empty and should be treated as such. Content type is irrelevant in this case.

> However, it doesn’t specify whether a content-type should still be included in such cases. 

Reviewer 2: In case of an empty payload, no content-type is required.

> However, it doesn’t specify whether a content-type should still be included in such cases. 

Reviewer 2: ```suggestion
In case of an empty payload, no content-type is required.
```
